### PR TITLE
Fix redraw event that causes inifinite loop

### LINKED ISSUE #255: linux: `request_redraw()` doesn't render the window or its widgets properly

### LINKED ISSUE #258: bug: `set_focus()` doesn't work properly on linux

### PR DESCRIPTION
**Pull request by wusyong**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [ ] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
Fix #255

### CODE REVIEW

**Comment by amrbashir** [comment]
> Fix #258

This is the wrong issue.

**Comment by wusyong** [comment]
@amrbashir I took the chance to refactor the event loop a bit. It looks nicer imho.

File: src/platform_impl/linux/event_loop.rs [hunk 1]
```diff
@@ -72,13 +72,21 @@ impl<T> EventLoopWindowTarget<T> {
   }
 }
 
+enum EventState {
+  NewStart,
+  EventQueue,
+  DrawQueue,
+}
```
Comment (amrbashir) [inline src/platform_impl/linux/event_loop.rs:79]: might be worth documenting what these event state means.

File: src/platform_impl/linux/event_loop.rs [hunk 2]
```diff
@@ -72,13 +72,21 @@ impl<T> EventLoopWindowTarget<T> {
   }
 }
 
+enum EventState {
+  NewStart,
+  EventQueue,
+  DrawQueue,
```
Comment (amrbashir) [inline src/platform_impl/linux/event_loop.rs]: might be worth documenting what they mean.

**Review by amrbashir — APPROVED** [review]

**Comment by amrbashir** [comment]
It is definitely better, gj as always

File: src/platform_impl/linux/event_loop.rs (hunk 2 above)
Comment (wusyong) [reply src/platform_impl/linux/event_loop.rs]: You are alright. Give me one more day. I'm gonna draw some sweet chart in the comment to demonstrate how the state flows.

Comment (amrbashir) [reply src/platform_impl/linux/event_loop.rs]: Wow, that would be brilliant.

**Review by amrbashir — APPROVED** [review]

**Pull request merged**